src/bundle/Controller/Content/ContentTreeController.php

```diff
@@ -110,6 +134,117 @@ public function loadSubtreeAction(Request $request): Root
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException
+     */
+    public function loadNodeExtendedInfoAction(Location $location): NodeExtendedInfo
+    {
+        $locationPermissionRestrictions = $this->getLocationPermissionRestrictions($location);
+
+        return new NodeExtendedInfo($locationPermissionRestrictions);
+    }
+
+    /**
+     * @return TPermissionRestrictions
+     *
+     * @throws \Ibexa\AdminUi\Exception\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\BadStateException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    public function getLocationPermissionRestrictions(Location $location): array
+    {
+        $lookupCreateLimitationsResult = $this->permissionChecker->getContentCreateLimitations($location);
+        $lookupUpdateLimitationsResult = $this->permissionChecker->getContentUpdateLimitations($location);
+
+        $createLimitationsValues = $this->lookupLimitationsTransformer->getGroupedLimitationValues(
+            $lookupCreateLimitationsResult,
+            [Limitation::CONTENTTYPE, Limitation::LANGUAGE]
+        );
+
+        $updateLimitationsValues = $this->lookupLimitationsTransformer->getGroupedLimitationValues(
+            $lookupUpdateLimitationsResult,
+            [Limitation::CONTENTTYPE, Limitation::LANGUAGE]
+        );
+
+        return [
+            'create' => [
+                'hasAccess' => $lookupCreateLimitationsResult->hasAccess(),
+                'restrictedContentTypeIds' => $createLimitationsValues[Limitation::CONTENTTYPE],
+                'restrictedLanguageCodes' => $createLimitationsValues[Limitation::LANGUAGE],
+            ],
+            'edit' => [
+                'hasAccess' => $lookupUpdateLimitationsResult->hasAccess(),
+                'restrictedLanguageCodes' => $updateLimitationsValues[Limitation::LANGUAGE],
+            ],
+            'delete' => [
+                'hasAccess' => $this->canUserRemoveContent($location),
+            ],
+            'hide' => [
+                'hasAccess' => $this->canUserHideContent($location),
+            ],
+        ];
+    }
+
+    /**
+     * @throws \Ibexa\AdminUi\Exception\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\BadStateException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     */
+    private function canUserRemoveContent(Location $location): bool
+    {
+        $content = $location->getContent();
+        $contentType = $content->getContentType();
+        $contentIsUser = (new ContentTypeIsUser($this->configResolver->getParameter('user_content_type_identifier')))
+            ->isSatisfiedBy($contentType);
+
+        $translations = $content->getVersionInfo()->languageCodes;
```

Comment:
Performance: magic getter.

For now we can do either:
```suggestion
        $translations = array_map(
            static fn (Language $language): string => $language->getLanguageCode(),
            $content->getVersionInfo()->getLanguages()
        );
```
or expand the scope of ibexa/core#349 and add `getLanguageCodes` method to `VersionInfo` abstract contracts.